src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationStrategies/DelegationValidationStrategy.cs

```diff
@@ -412,12 +413,24 @@ protected bool IsValidNode(TexlNode node, TexlBinding binding)
             return true;
         }
 
-        private bool IsUserCallNodeDelegable(TexlNode node, TexlBinding binding)
+        protected bool IsValidDelegatableAsyncNode(TexlNode node, TexlBinding binding)
         {
-            if ((node is DottedNameNode)
-                && (node.AsDottedName().Left is CallNode)
-                && (binding.GetInfo(node.AsDottedName().Left.AsCall()).Function is ICustomDelegationFunction customDelFunc)
-                && customDelFunc.IsUserCallNodeDelegable())
+            // Dotted Name node with left hand delegatable async call
+            if ((node is DottedNameNode) && (node.AsDottedName().Left is CallNode))
```

Comment:
>(node is DottedNameNode) && (node.AsDottedName().Left is CallNode) [](http://example.com/codeflow?start=16&length=66)

This should be handled in IsValidDottedNameNode in DottedNamedValidationStrategy.

If LHS is call node then you just need to call IsValidCallNode on dottednamenode.Left and handle this there.